inc/saihostintf.h

```diff
@@ -321,19 +349,38 @@ typedef enum _sai_hostif_trap_attr_t
     SAI_HOSTIF_TRAP_ATTR_START,
 
     /**
+    * @brief Is user defined trap
+    *
+    * @type bool
+    * @flags CREATE_ONLY
+    * @default false
+    */
+    SAI_HOSTIF_TRAP_ATTR_IS_USER_DEFINED_TRAP = SAI_HOSTIF_TRAP_ATTR_START,
```

Comment:
What was the motivation to move user defined traps into the separate enum? Does it limits somehow the list of supported SAI_HOSTIF_TRAP_ATTRs by user defined traps? E.g., is it valid to use EXCLUDE_PORT_LIST or GROUP for user defined traps?